### ISSUE TITLE
logs are not getting pushed in report portal

### ISSUE THREAD

**Issue by nikhilsecret** [reportportal/reportportal#2181]
logs are printed on IntellJ console but they are not pushed in report portal.

i am using below log4j2.xml

`<?xml version="1.0" encoding="UTF-8"?>
<Configuration status="WARN">
    <Appenders>
        <Console name="ConsoleAppender" target="SYSTEM_OUT">
            <PatternLayout
                    pattern="%d [%t] %-5level %logger{36} - %msg%n%throwable"/>
        </Console>
        <ReportPortalLog4j2Appender name="ReportPortalAppender">
            <PatternLayout
                    pattern="%d [%t] %-5level %logger{36} - %msg%n%throwable"/>
        </ReportPortalLog4j2Appender>
    </Appenders>
    <Loggers>
        <Root level="INFO">
            <AppenderRef ref="ConsoleAppender"/>
            <AppenderRef ref="ReportPortalAppender"/>
        </Root>
        <Logger name="com.epam.reportportal" level="INFO" additivity="false">
            <AppenderRef ref="ConsoleAppender"/>
            <AppenderRef ref="ReportPortalAppender"/>
        </Logger>
        <Logger name="binary_data_logger" level="DEBUG" additivity="false">
            <AppenderRef ref="ReportPortalAppender"/>
        </Logger>
    </Loggers>
</Configuration>
`

also dependencies i am using 

`        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-api</artifactId>
            <version>2.0.7</version>
        </dependency>

        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-simple</artifactId>
            <version>2.0.7</version>
        </dependency>

        <dependency>
            <groupId>org.apache.logging.log4j</groupId>
            <artifactId>log4j-slf4j-impl</artifactId>
            <version>2.17.2</version>
        </dependency>
         <dependency>
            <groupId>com.epam.reportportal</groupId>
            <artifactId>agent-java-testng</artifactId>
            <version>5.3.1</version>
        </dependency>

        <dependency>
            <groupId>com.epam.reportportal</groupId>
            <artifactId>logger-java-log4j</artifactId>
            <version>5.1.8</version>
            <scope>test</scope>
        </dependency>
        
        <dependency>
            <groupId>com.epam.reportportal</groupId>
            <artifactId>logger-java-rest-assured</artifactId>
            <version>5.2.4</version>
        </dependency>

        <dependency>
            <groupId>com.squareup.okhttp3</groupId>
            <artifactId>okhttp</artifactId>
            <version>3.14.9</version>
        </dependency>
        
        `

**Comment by nikhilsecret** [comment]
@HardNorth  i initialised the logger like this -

private static final Logger LOGGER = LoggerFactory.getLogger(Execute_CreateResource.class);
 and i am using LOGGER like this inside my testNg method- 
 LOGGER.info("macAddress of resource is " + macAddress.get());
 
 
 
![Screenshot 2023-11-17 at 7 27 54 PM](https://github.com/reportportal/reportportal/assets/96907969/6af955d5-dc55-4c1d-8fc0-0e5918e255c3)


console output look like this-

![Screenshot 2023-11-17 at 7 28 30 PM](https://github.com/reportportal/reportportal/assets/96907969/011f4e73-70f7-4132-8ad0-9fcafc2b5b8a)


please let me know if we can have a session on this, i need to integrate report portal asap.

**Comment by HardNorth** [comment]
@nikhilsecret Your tests do not pick your Log4j configuration, since the format of log lines is different, by the configuration it should look like that:
`2023-11-17 17:21:28,112 [TestNG-test=Test data provider with factory-1] INFO  com.epam.reportportal.example.testng.log4j.parametrized.ParametrizedTest - Hero{name='Flash', power='Speed', race='Human'}`

But your output differs.

I believe it picks logging binder from `slf4j-simple` dependency and doesn't look for different providers. But our examples use `slf4j-api` like here:
https://github.com/reportportal/examples-java/blob/master/example-testng-log4j/pom.xml#L45

So that not actually related to RP or its libraries, please update your configuration.

**Comment by nikhilsecret** [comment]
@HardNorth  if i don't use slf4j-simple, then it is not event printing the logs in intellJ console.

**Comment by HardNorth** [comment]
@nikhilsecret Yes, because you Log4j not even initialize. `slf4j-simple` is the simplest console logging without any configuration options, you need to properly configure more complex library like `Log4j` or `logback` to have an options to upload logs to ReportPortal. Or to even customize your log output format. But this is besides the area of ReportPortal responsibilities, you have to be able to configure your logging framework yourself.

Well, our I believe our documentation covers that completely, just read it carefully: https://github.com/reportportal/logger-java-log4j
